apis/v1alpha1/destination.go

```diff
@@ -28,6 +28,9 @@ type Destination struct {
 	// +optional
 	Ref *corev1.ObjectReference `json:"ref,omitempty"`
 
+	// +optional
+	DeprecatedObjectReference *corev1.ObjectReference `json:",inline"`
```

Comment:
with this change, can you add some json marshaling tests to confirm we can still support both formats?

Reply:
Do you mean writing unit tests which can accept parameters like  Kubernetes YAML and we verify we can still convert it to a Destination object?